kong/api/init.lua

```diff
@@ -54,6 +54,7 @@ local function on_error(self)
       return responses.send_HTTP_BAD_REQUEST(err.tbl or err.message)
     end
   end
+  return responses.send_HTTP_BAD_REQUEST(tostring(err))
```

Comment:
Because we are outside of the `type(err) == "table"` condition here, do we really need to invoke `tostring` here?

Reply:
As I see it it's maybe not needed in that particular case, but as in all usage of the send_HTTP_* functions strings are passed, I thought it's better for the consistency of how we use them. 

Also, doc at kong/tools/responses.lua:183 suggests only strings or tables are expected.

As from the practical point of view, it doesn't matter much, ping again if you think `tostring` should be removed.

Reply:
@thibaultcha We've had some bugs in the past where the couldn't get the underlying error because the error handler failed exactly on a missing `tostring`, which obfuscated the underlying error. Since this isn't a hot code path, I'd say better-safe-than-sorry in this case. My 2cts.

Reply:
I doubt that. I checked the `responses` module prior to posting that comment, and in the module we can observe that the argument gets logged with `tostring()`, or send in the response body by deferring its encoding to lua-cjson (https://github.com/Kong/kong/blob/master/kong/tools/responses.lua#L141-L142), so there is no reason for calling `tostring()` here, except cluttering the code as more and more people keep copying the same snippets of code all over the place, as is often the case.

Reply:
I removed the tostring part.  Tell me if I should modify something else, thanks!

Reply:
@thibaultcha my remark was not related to the specific call. At the time it was a concatenation done before calling the responses method. All I'm saying is that we should make the error handling code path as resilient as possible to prevent errors from happening when we're handling errors.

@kidd thx for the update

Reply:
> All I'm saying is that we should make the error handling code path as resilient as possible to prevent errors from happening when we're handling errors.

Most definitely.